### PR TITLE
Change compress to work mainly with Buffers

### PR DESCRIPTION
**Pull request by sigvef**

### CODE REVIEW

**Comment by stianjensen** [comment]
"I'm a scientist, I work mainly with buffers"

**Comment by stianjensen** [comment]
Trying to test this, but can't currently get any demos compiled on neither the master branch nor here on my machine...

**Comment by sigvef** [comment]
Does your project have a res/camerapaths.json?

**Comment by sigvef** [comment]
Also, projects *must* have a `lib/` folder (until somebody makes it optional :P ).

**Comment by stianjensen** [comment]
Yup, it has camerapaths.json and lib (but I'll definitely make lib optional).
I opened another issue about this: #216

**Comment by sigvef** [comment]
ping, rebased

File: nin/backend/compress.js [hunk 1]
```diff
@@ -89,20 +93,28 @@ function compress(projectPath, payload, callback) {
...
+    var chunks = [], i = 0, n = buffer.length;
+    while (i < n) { chunks.push(buffer.slice(i, i += len)); }
+    return chunks;
+  }
 
-  var scanlines = payload.match(new RegExp('[\\s\\S]{1,' + (width * 3) + '}', 'g'));
-  var scanlinesBuffer = Buffer.concat(scanlines.map(function(scanline){
-    return new Buffer('\0' + scanline, 'binary');
+  var scanlines = bufferChunk(payload, width * 3);
+  var total = 0;
```
Comment (stianjensen) [inline nin/backend/compress.js]: Where's this variable used?

Comment (sigvef) [inline nin/backend/compress.js]: Nowhere, it seems. Removed.

**Pull request merged**